### PR TITLE
Creating_unit_test_and_methods_for_place_page

### PR DESCRIPTION
**Pull request by RoxVer**

### CODE REVIEW

File: src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        return new ResponseEntity<>(place, HttpStatus.OK);
+    }
+    
+    @RequestMapping(value = "/getImage", method = RequestMethod.GET)
+    public void showImage(HttpServletResponse response) throws Exception {
+
+      ByteArrayOutputStream jpegOutputStream = new ByteArrayOutputStream();
+
+      try {
+        BufferedImage image = ImageIO.read(new File("C:/dev/projects/eclipse-projects/GreenTourism3/src/main/webapp/images/Some_Image.jpg"));
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java:81]: This is hard coding! You can't do this.
And, by the way, our linux server will not have this path for cure!

File: src/main/java/ua/softserve/rv_018/greentourism/model/Place.java [hunk 2]
```diff
@@ -24,6 +24,12 @@
 	@Column(name = "description")
 	private String description;
 	
+	@Column(name = "otherInfo")
+	private String otherInfo;
+	
+	@Column(name = "feedbacks")
+	private String feedbacks;
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/model/Place.java:31]: Do we really have this two fields in our Place model?

File: src/main/java/ua/softserve/rv_018/greentourism/repository/PlaceRepository.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package ua.softserve.rv_018.greentourism.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import ua.softserve.rv_018.greentourism.model.Place;
+
+@Repository
+public interface PlaceRepository extends JpaRepository<Place, Long> {
+	Place findById(Long id);
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/repository/PlaceRepository.java:10]: You do not need to specify this method, Spring data jpa provides it for you.

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package ua.softserve.rv_018.greentourism.service;
+
+import ua.softserve.rv_018.greentourism.model.Place;
+
+public interface PlaceService {
+	
+    Place findOnePlace(Long id);
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java:7]: Plz, refer to the user implementation.
It's better to call it findOne

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    @Autowired
+    private PlaceRepository placeRepository;
+
+	@Override
+	public Place findOnePlace(Long id) {
+		logger.info("> Place findOnePlace id:{}", id);
+
+        Place place = placeRepository.findById(id);
+
+        logger.info("< Place findOnePlace id:{}", id);
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java:35]: It seems like the problem with code format here.

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+     * The Spring Data repository for Role entities.
+     */
+    @Autowired
+    private PlaceRepository placeRepository;
+
+	@Override
+	public Place findOnePlace(Long id) {
+		logger.info("> Place findOnePlace id:{}", id);
+
+        Place place = placeRepository.findById(id);
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java:33]: Plz, refer to the user implementation

File: src/test/java/ua/softserve/rv_018/greentourism/controller/PlaceControllerUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+public class PlaceControllerUnitTest {
+    public static final String EMPTY_COLLECTION = "[]";
+    public static final String EMPTY_VALUE = "";
+    public static final String VALUE ="{\"id\":1,\"langtitude\":\"langtitude\",\"longtitude\":\"longtitude\"}";
+    public static final Point PLACE = new Point(1l, "langtitude", "longtitude");
+    public static final String HEADER_LOCATION = "http://localhost/place/1";
```
Comment (Sabfir) [inline src/test/java/ua/softserve/rv_018/greentourism/controller/PlaceControllerUnitTest.java:28]: Plz, check all this fields and delete those you don't need

File: src/test/java/ua/softserve/rv_018/greentourism/controller/PlaceControllerUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    private MockMvc mockMvc;
+
+    @InjectMocks
+    private PlaceController placeController;
+
+    @Mock
+    private PlaceService placeService;
+    
+    @Mock
+    private HttpHeaders httpHeaders;
```
Comment (Sabfir) [inline src/test/java/ua/softserve/rv_018/greentourism/controller/PlaceControllerUnitTest.java:39]: Do you really need to mock this one?

File: src/test/java/ua/softserve/rv_018/greentourism/controller/PlaceControllerUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        MockitoAnnotations.initMocks(this);
+        
+        mockMvc = MockMvcBuilders.standaloneSetup(this.placeController).build();
+    }
+    
+    @Test
+    public void testGetPlace() throws Exception {
+    	Mockito.when(placeService.findOnePlace(1l)).thenReturn(PLACE);
+        
+        mockMvc.perform(get("/place/1"))
```
Comment (Sabfir) [inline src/test/java/ua/softserve/rv_018/greentourism/controller/PlaceControllerUnitTest.java:52]: Plz, do not use hard coded variables

File: src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java (hunk 1 above)
Comment (RoxVer) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java:81]: It was for manual testing. 
So which path we will have (u said in slack that we have this path)?

File: src/main/java/ua/softserve/rv_018/greentourism/model/Place.java (hunk 2 above)
Comment (RoxVer) [inline src/main/java/ua/softserve/rv_018/greentourism/model/Place.java:31]: According to Misha's "print-screen" pic, we'll have these fields on our Place page.

Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/model/Place.java:31]: Creating model, plz refer to the db blueprint.
We don't have this fields int the db table place

File: src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java (hunk 1 above)
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java:81]: I have never said about this nothing "C:/dev/projects/eclipse-projects/GreenTourism3"
Do you really think everyone included our servers will have this path?

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java (hunk 4 above)
Comment (RoxVer) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java:7]: What's the difference how to call it?

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java (hunk 6 above)
Comment (RoxVer) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java:33]: Why should I?

File: src/test/java/ua/softserve/rv_018/greentourism/controller/PlaceControllerUnitTest.java (hunk 9 above)
Comment (RoxVer) [inline src/test/java/ua/softserve/rv_018/greentourism/controller/PlaceControllerUnitTest.java:52]: hard coded variables are needed for unit tests

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java (hunk 4 above)
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java:7]: Have you conspired with Max?

Comment (RoxVer) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java:7]: what??

File: src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java (hunk 1 above)
Comment (RoxVer) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java:81]: yeah, I know, it was for manual testing, I'll change it

Comment (RoxVer) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java:81]: Also, it doesn't work with this path:  /src/main/webapp/images/

Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java:81]: I can give you direct answer: "/images/blabla.png"
But it's important to understand why is that?!
Folder webapp is the root of the classpath. Knowing it you can guess you should start from "/images....".
Even if you don't know it, here is the other way: you can build your project, creating war archive and find out where is the image folder placed in the archive.
1. open terminal, go to the project root
2. execute command: mvn clean package
3. go to the target folder, you will find war archive
4. open war archive, the images folder is in the top level of the archive, so the path to the folder should be "/images...."

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java (hunk 4 above)
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java:7]: Plz, just change the name, I'll explain you in the daily meeting why.

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java (hunk 6 above)
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java:33]: Plz, just change the name of the method, I'll explain you in the daily meeting why.

File: src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java (hunk 1 above)
Comment (RoxVer) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java:81]: "Knowing it you can guess you should start from "/images...."." - it didn't work 

I'll try these steps.

File: src/main/java/ua/softserve/rv_018/greentourism/model/Place.java (hunk 2 above)
Comment (RoxVer) [reply src/main/java/ua/softserve/rv_018/greentourism/model/Place.java:31]: So, where do we plan to store feedbacks and otherInfo then?

File: src/main/java/ua/softserve/rv_018/greentourism/repository/PlaceRepository.java (hunk 3 above)
Comment (RoxVer) [reply src/main/java/ua/softserve/rv_018/greentourism/repository/PlaceRepository.java:10]: It's not working otherwise

**Pull request closed without merging**